.github/workflows/check-links.yaml

```diff
@@ -0,0 +1,51 @@
+name: check-links
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  changedfiles:
+    name: changed files
+    runs-on: ubuntu-latest
+    env:
+      PR_HEAD: ${{ github.event.pull_request.head.sha }}
+    outputs:
+      md: ${{ steps.changes.outputs.md }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changes
+        run: |
+          echo "md=$(git diff --name-only --diff-filter=ACMRTUXB $(git merge-base origin/main $PR_HEAD) $PR_HEAD | grep .md$ | xargs)" >> $GITHUB_OUTPUT
+
+  check-links:
+    runs-on: ubuntu-latest
+    needs: changedfiles
+    if: ${{needs.changedfiles.outputs.md}}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 0
+
+      - name: Install markdown-link-check
+        run: npm install
+
+      - name: Run markdown-link-check
+        run: |
+          npx --no -- markdown-link-check \
+            --verbose \
+            --config .github/workflows/check_links_config.json \
+            ${{needs.changedfiles.outputs.md}} \
+            || { echo "Check that anchor links are lowercase"; exit 1; }
```

Comment:
Could you explain this line a bit?

Reply:
Ok,  This line of code runs the markdown-link-check with verbose logging enabled and specifies the custom configuration file (check_links_config.json). The `needs.changedfiles.outputs.md` variable is used to dynamically target only the markdown files that were modified in the PR.

The || `{ echo "Check that anchor links are lowercase"; exit 1; } ` part is a fallback check to ensure that if there are any issues with the anchor links not being lowercase (which is a requirement for valid markdown links), the workflow will fail and print a meaningful message for the contributor to resolve. 

However, if this check is redundant, I can adjust or remove it.

Reply:
Not objecting to the current implementation, but I wonder if maybe a linter would do a better job of that particular edge case.  What do you think?

Reply:
I agree that a linter might be more appropriate for this edge case, as it would likely cover more general cases of markdown formatting and anchor validation. Not quite sure how to implement exactly for this as I'm new here. I'll appreciate any help @tylerbenson

Reply:
I don't have any specific insight here.  Maybe investigate what other OpenTelemetry repos are doing?